astropy/table/table.py

```diff
@@ -1000,6 +1000,60 @@ def pprint(self, max_lines=None, max_width=None, show_name=True,
             else:
                 print(line)
 
+    def pprint_all(self, max_lines=-1, max_width=-1, show_name=True,
+               show_unit=None, show_dtype=False, align=None):
+        """Print a formatted string representation of the entire table.
+
+        If no value of ``max_lines`` is supplied then the height of the
+        screen terminal is used to set ``max_lines``.  If the terminal
+        height cannot be determined then the default is taken from the
+        configuration item ``astropy.conf.max_lines``.  If a negative
+        value of ``max_lines`` is supplied then there is no line limit
+        applied.
+
+        The same applies for max_width except the configuration item is
+        ``astropy.conf.max_width``.
+
+        Parameters
+        ----------
+        max_lines : int
+            Maximum number of lines in table output.
+
+        max_width : int or `None`
+            Maximum character width of output.
+
+        show_name : bool
+            Include a header row for column names. Default is True.
+
+        show_unit : bool
+            Include a header row for unit.  Default is to show a row
+            for units only if one or more columns has a defined value
+            for the unit.
+
+        show_dtype : bool
+            Include a header row for column dtypes. Default is True.
+
+        align : str or list or tuple or `None`
+            Left/right alignment of columns. Default is right (None) for all
+            columns. Other allowed values are '>', '<', '^', and '0=' for
+            right, left, centered, and 0-padded, respectively. A list of
+            strings can be provided for alignment of tables with multiple
+            columns.
+        """
+        lines, outs = self.formatter._pformat_table(self, max_lines, max_width,
```

Comment:
All this code should just be:
```
return self.pprint(max_lines, max_width, ...)
```
I.e. delegate to the original method.  One could also imagine doing some hackery with the `__doc__` values to avoid repetition of the docstring info, but that's not necessary.